pkg/virt-operator/application.go

```diff
@@ -297,3 +351,58 @@ func (app *VirtOperatorApp) AddFlags() {
 
 	app.AddCommonFlags()
 }
+
+func (app *VirtOperatorApp) getSelfSignedCert() error {
+	var ok bool
+
+	caKeyPair, _ := triple.NewCA("kubevirt.io")
+	keyPair, _ := triple.NewServerKeyPair(
+		caKeyPair,
+		"kubevirt-operator-webhook."+app.operatorNamespace+".pod.cluster.local",
+		"kubevirt-operator-webhook",
+		app.operatorNamespace,
+		"cluster.local",
+		nil,
+		nil,
+	)
+
+	secret := &k8sv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      virtOperatorCertSecretName,
+			Namespace: app.operatorNamespace,
+			Labels: map[string]string{
+				v1.AppLabel: "virt-operator-webhooks",
+			},
+		},
+		Type: "Opaque",
+		Data: map[string][]byte{
+			certBytesValue:        cert.EncodeCertPEM(keyPair.Cert),
+			keyBytesValue:         cert.EncodePrivateKeyPEM(keyPair.Key),
+			signingCertBytesValue: cert.EncodeCertPEM(caKeyPair.Cert),
+		},
+	}
+	_, err := app.clientSet.CoreV1().Secrets(app.operatorNamespace).Create(secret)
+	if errors.IsAlreadyExists(err) {
```

Comment:
We had the before. The background is that we avoid this way race-condition. Otherwise we would have to do GET, CREATE (here we are again racing with others), GET.

Reply:
Makes sense. Thanks for the explanation!